lib/semantics/resolve-names.cc

```diff
@@ -313,6 +315,9 @@ class ArraySpecVisitor {
 // Manage a stack of Scopes
 class ScopeHandler : public ImplicitRulesVisitor {
 public:
+  template<typename T> void Walk(const T &);
+  void set_this(ResolveNamesVisitor *x) { this_ = x; }
```

Comment:
this and this_ look very similar.

Reply:
They are meant to. They have the same value but different types. This is to avoid having to cast `this` to `ResolveNamesVisitor *` when calling `Walk`.